packages/svelte/src/internal/client/dom/elements/bindings/this.js

```diff
@@ -22,14 +22,14 @@ function is_bound_this(bound_value, element_or_component) {
  * @returns {void}
  */
 export function bind_this(element_or_component, update, get_value, get_parts) {
-	effect(() => {
+	render_effect(() => {
 		/** @type {unknown[]} */
 		var old_parts;
 
 		/** @type {unknown[]} */
 		var parts;
 
-		render_effect(() => {
+		pre_effect(() => {
```

Comment:
turns out we can revert this part, it makes no difference

```suggestion
		render_effect(() => {
```